### PR TITLE
borgversion placeholder

### PR DESCRIPTION
**Pull request by leo-b**
As proposed in #1392:
This pull request introduces a new placeholder {borgversion} and adds placeholder substitution to --remote-path and the BORG_RSH environment.
It also adds a BORG_VERSION environment variable before running the ssh command.

### CODE REVIEW

**Comment by ThomasWaldmann** [comment]
discussion: put this into 1.0-maint?

imho, version management makes more sense if more versions support it.

**Comment by PlasmaPower** [comment]
@ThomasWaldmann Yes I this should also definitely be in 1.0-maint.

**Comment by leo-b** [comment]
I'd also vote for inclusion in 1.0-maint... Changes are minimally invasive and those, who are about to establish a new backup infrastructure with borg, need a version selection as soon as possible, or they have to build some cumbersome workaround anyway.

**Comment by enkore** [comment]
+1

File: docs/usage/help.rst.inc [hunk 1]
```diff
@@ -32,6 +33,10 @@ Repository (or Archive) URLs and --prefix values support these placeholders:
 
     The current process ID.
 
+{borgversion}
+
+    The version of borg.
+
```
Comment (enkore) [inline docs/usage/help.rst.inc]: When you make a new PR against 1.0-maint, please bear in mind that

> IMPORTANT: this file is auto-generated from borg's built-in help, do not edit!

Ie. these changes should go in the docstrings in borg.archiver

**Comment by ThomasWaldmann** [comment]
as the PR against 1.0-maint is merged now, this one is maybe not needed any more as we'll just merge 1.0-maint into master later.

**Comment by leo-b** [comment]
OK thanks, I'll close it.

**Pull request closed without merging**